zerver/views/messages.py

```diff
@@ -308,6 +313,22 @@ def _by_search_tsearch(self, query, operand, maybe_negate):
         cond = column("search_tsvector").op("@@")(tsquery)
         return query.where(maybe_negate(cond))
 
+def highlight_match_tag_pos(text, pos, loc, tag_open, tag_close):
+    # type: (AnyStr, int, Tuple[int, int], AnyStr, AnyStr) -> Tuple[int, int]
+
+    (offset, length) = loc
+
+    # When the offset is inside a tag, we want to move it back, so that it
+    # highlights the whole tag, not just some attribute of it
+    last_tag_open = text.rfind(tag_open, pos, offset+length)
+    last_tag_close = text.rfind(tag_close, pos, offset)
+    if last_tag_close < last_tag_open:
+        next_tag_close = text.find(tag_close, offset+length-1)
+        offset = last_tag_open
+        length = next_tag_close - last_tag_open + 1
+
+    return (offset, length)
```

Comment:
Yeah, going to write a test for that one :+1: 

The `tag_open` and `tag_close` were made into arguments since python's `rfind` didn't like different encodings at all (using `'>'`/`u'>'` failed with bytes, using `b'>'` failed with unicode). So, yeah, seems like nothing is "just an ASCII character" in python :|